AutomatedTesting/Gem/PythonTests/EditorPythonTestTools/editor_python_test_tools/script_canvas_tools_qt.py

```diff
@@ -0,0 +1,448 @@
+
+    returns: boolean value of the search attempt
+    """
+    self.node_tree_search_box.setText(node_name)
+    helper.wait_for_condition(lambda: self.node_tree_search_box.text() == node_name, WAIT_TIME_3)
+    # Try clicking ENTER in search box multiple times
+    found_node = False
+    for _ in range(number_of_retries):
+        QtTest.QTest.keyClick(self.node_tree_search_box, QtCore.Qt.Key_Enter, QtCore.Qt.NoModifier)
+        found_node = helper.wait_for_condition(
+            lambda: pyside_utils.find_child_by_pattern(self.node_tree_view, {"text": node_name}) is not None, WAIT_TIME_3)
+        if found_node is True:
+            break
+    return found_node
+
+def get_node_palette_node_tree_qt_object (self):
+    """
+    function for retrieving the tree view qt object for the node palette
+
+    params self: the script calling this function
+
+    returns: a tree view qt object
+    """
+    node_palette_widget = self.sc_editor.findChild(QtWidgets.QDockWidget, NODE_PALETTE_QT)
+    node_palette_node_tree = node_palette_widget.findChild(QtWidgets.QTreeView, TREE_VIEW_QT)
+    return node_palette_node_tree
+
+
+def get_node_palette_category_qt_object(self, category_name):
+    """
+    function for retrieving the qt object for a node palette category
+
+    param self: the script calling this function
+    param category_name: string for the category label you are searching node palette for
+
+    returns: the qt object for the node palette category
+    """
+    node_palette_node_tree = get_node_palette_node_tree_qt_object(self)
+    node_palette_category = pyside_utils.find_child_by_pattern(node_palette_node_tree, category_name)
+    return node_palette_category
+
+def get_node_inspector_node_titles(self, sc_graph_node_inspector, sc_graph):
+    """
+    function for retrieving the node inspector's node titles from all nodes in a script canvas graph. function takes
+    a script canvas graph and node inspector qt widget.
+
+    param self: the script calling this function
+    param sc_graph_node_inspector: the sc graph node inspector qt widget
+    param sc_graph: the sc graph qt widget
+
+    returns: a list of node titles (i.e Print - Utilities/Debug). If there are duplicates of a node then the title
+    will include ( X Selected) in the string.
+    """
+    node_inspector_scroll_area = sc_graph_node_inspector.findChild(QtWidgets.QScrollArea, "")
+    # perform ctrl+a keystroke to highlight all nodes on the graph
+    QtTest.QTest.keyClick(sc_graph, "a", Qt.ControlModifier, WAIT_FRAMES_200)
+    node_inspector_backgrounds = node_inspector_scroll_area.findChildren(QtWidgets.QFrame, "Background")
+    titles = []
+    for background in node_inspector_backgrounds:
+        background_title = background.findChild(QtWidgets.QLabel, NODE_INSPECTOR_TITLE_KEY)
+        if background_title.text() != "":
+            titles.append(background_title.text())
+    return titles
+
+
+def get_sc_editor_node_inspector(sc_editor):
```

Comment:
Can this be refactorted into SC Editor or its own QtPy class?

Reply:
getters are "not pythonic" but yes there will be a reference to the node inspector on the sc editor class in the future.